exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SQLExpressionBuilder.kt

```diff
@@ -272,46 +288,111 @@ interface ISqlExpressionBuilder {
         return NeqOp(this, wrap(entityID))
     }
 
+    /** Checks if this [EntityID] expression is not equal to some [other] expression. */
+    infix fun <T : Comparable<T>, E : EntityID<T>?, V : T?> ExpressionWithColumnType<E>.neq(
+        other: Expression<in V>
+    ): Op<Boolean> = when (other as Expression<*>) {
+        is Op.NULL -> isNotNull()
+        else -> NeqOp(this, other)
+    }
+
+    /** Checks if this expression is not equal to some [other] [EntityID] expression. */
+    infix fun <T : Comparable<T>, V : T?, E : EntityID<T>?> Expression<V>.neq(
+        other: ExpressionWithColumnType<in E>
+    ): Op<Boolean> = other neq this
+
+    // LESS THAN
+
     /** Checks if this expression is less than some [t] value. */
+    @LowPriorityInOverloadResolution
     infix fun <T : Comparable<T>, S : T?> ExpressionWithColumnType<in S>.less(t: T): LessOp = LessOp(this, wrap(t))
```

Comment:
As was already done for `eq()` and `neq()` above, this is necessary to ensure the right overload is used.